google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAdCustomizer.java

```diff
@@ -0,0 +1,498 @@
+import com.google.ads.googleads.v2.services.GoogleAdsServiceClient.SearchPagedResponse;
+import com.google.ads.googleads.v2.services.MutateAdGroupAdResult;
+import com.google.ads.googleads.v2.services.MutateAdGroupAdsResponse;
+import com.google.ads.googleads.v2.services.MutateFeedItemResult;
+import com.google.ads.googleads.v2.services.MutateFeedItemTargetsResponse;
+import com.google.ads.googleads.v2.services.MutateFeedItemsResponse;
+import com.google.ads.googleads.v2.services.MutateFeedMappingsResponse;
+import com.google.ads.googleads.v2.services.MutateFeedsResponse;
+import com.google.ads.googleads.v2.services.SearchGoogleAdsRequest;
+import com.google.ads.googleads.v2.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.joda.time.DateTime;
+
+/** Adds an ad customizer feed and associates it with the customer. Then it adds an ad
+ * that uses the feed to populate dynamic data. */
+public class AddAdCustomizer {
+
+  /** We're doing only searches by resource_name in this example, we can set page size = 1. */
+  private static final int PAGE_SIZE = 1;
+
+  private static class AddAdCustomizerParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_GROUP_ID, required = true)
+    private List<Long> adGroupIds;
+
+    @Parameter(names = ArgumentNames.FEED_NAME, required = true)
+    private String feedName;
+  }
+
+  public static void main(String[] args) throws IOException {
+    AddAdCustomizerParams params = new AddAdCustomizerParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.adGroupIds = Arrays.asList(
+          Long.parseLong("INSERT_AD_GROUP_ID_HERE"),
+          Long.parseLong("INSERT_AD_GROUP_ID_HERE"));
+      params.feedName = "INSERT_FEED_NAME_HERE";
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddAdCustomizer().runExample(
+          googleAdsClient, params.customerId, params.adGroupIds, params.feedName);
```

Comment:
Pass the params object directly to runExample. The examples CI will shortly require this.

Reply:
Done.